src/main/scala/org/allenai/deep_qa/data/BackgroundCorpusSearcher.scala

```diff
@@ -82,19 +82,39 @@ class LuceneBackgroundCorpusSearcher(params: JValue) extends BackgroundCorpusSea
   lazy val settings = Settings.builder().put("cluster.name", esClusterName).build()
   lazy val esClient = TransportClient.builder().settings(settings).build().addTransportAddress(address)
 
-  override def getBackground(query: String): Seq[String] = {
+  override def getBackground(query: Query): Seq[String] = {
+    // Make the bool query and the more-or-less canonical query string for the
+    // consolidation step (below)
+    val queryBuilder = QueryBuilders.boolQuery()
+    var queryString = ""
+    query match {
+      case sq: StringQuery => {
+        queryBuilder.should(QueryBuilders.matchQuery("text", sq.query))
+        queryString = sq.query
+      }
+      case bq: BoostedQuery => {
+        for ((qString, boostAmt) <- bq.boostedQueries) {
+          queryBuilder.should(QueryBuilders.matchQuery("text", qString).boost(boostAmt))
+          queryString = bq.boostedQueries.map(q => q._1).mkString(" ")
```

Comment:
The `q` inside a very short call to `map()` like this is one place where I make exceptions to the rule against abbreviations.  However, in this case you don't even need it at all, as you can just do `boostedQueries.map(_._1).mkString(" ")`.